File: Frontend/src/app/models/journal/journal.model.ts
import {Student} from "../account/student";
import {Lesson} from "../shedule/lesson";
import {Pair} from "../shedule/pair";

export class Journal {
   public students: Array<Student>;
   public dates: Date[];
   public journalCell: Array<JournalCell>;
   public lesson: Lesson;
   public pairs: Array<Pair>;
   constructor() {
      this.students = new Array();
      this.dates = [];
      this.journalCell = new Array();
      this.pairs = new Array();
   }
}

export class JournalCell {

   public id: number;
   public value: number;
   public student: Student;
    public date: Date;
    public type: PointType;
    public pair: Pair;

    constructor() {
        this.id = 0;
        this.value = 0;
        this.student = new Student();
        this.type = new PointType();
        this.pair = new Pair();
    }
}


export class PointType {

    public id: number;
    public name: String;

    constructor() {
        this.id = 0;
        this.name = "";
    }

}

File: Backend/src/main/java/com/unesco/core/models/journal/JournalModel.java
package com.unesco.core.models.journal;

import com.unesco.core.models.account.StudentModel;
import com.unesco.core.models.shedule.LessonModel;
import com.unesco.core.models.shedule.PairModel;

import java.util.Date;
import java.util.List;

public class JournalModel
{
   private LessonModel lesson;
   private List<PairModel> pairs;
   private List<StudentModel> students;
   private List<Date> dates;
   private List<PointModel> journalCell;

   public JournalModel()
   {
   }

   public LessonModel getLesson() {
      return lesson;
   }
   public void setLesson(LessonModel lesson) {
      this.lesson = lesson;
   }

   public List<StudentModel> getStudents() {
      return students;
   }
   public void setStudents(List<StudentModel> students) {
      this.students = students;
   }

   public List<Date> getDates() {
      return dates;
   }
   public void setDates(List<Date> dates) {
      this.dates = dates;
   }

   public List<PointModel> getJournalCell() {
      return journalCell;
   }
   public void setJournalCell(List<PointModel> journalCell) {
      this.journalCell = journalCell;
   }

   public List<PairModel> getPairs() {
      return pairs;
   }
   public void setPairs(List<PairModel> pairs) {
      this.pairs = pairs;
   }

}
